tools/dreport.d/ibm.d/gendumpheader

```diff
@@ -50,6 +50,31 @@ function add_null() {
     printf '%*s' $a | tr ' ' "\0" >> $FILE
 }
 
+# Function to add Originator details to dump header
+function add_originator_details() {
+    if [ -z "$ORIGINATOR_TYPE" ]; then
+        add_null 4
+        return
+    fi
+    printf '%s' "$ORIGINATOR_TYPE" >> "$FILE"
+    len=${#ORIGINATOR_TYPE}
+    nulltoadd=$(( SIZE_4 - len ))
+    if [ "$nulltoadd" -ne 0 ]; then
+        add_null "$nulltoadd"
+    fi
+
+    if [ -z "$ORIGINATOR_ID" ]; then
```

Comment:
may be adding a echo statement with why we are adding these null bytes would help better debuggability later?
same for the earlier if statement as well.

Reply:
Would have love to do so but for other fields also it is the same way. Min tracing